state-synchronizer/src/coordinator.rs

```diff
@@ -1221,6 +1221,15 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             }
         };
 
+        let target_version = target
+            .version()
+            .unwrap_or_else(|| known_version.wrapping_add(1));
+        counters::set_version(counters::VersionType::Target, target_version);
+        let highest_version = self
+            .pending_ledger_infos
+            .highest_version()
+            .unwrap_or(target_version);
```

Comment:
Nit: I don't have a lot of scope into how these metrics are used, but would we rather not just ".unwrap_or(known_version)"? e.g., if pending ledger infos is empty and there is no target, highest_version == target_version, so we lose some information in the metrics about the latest known_version? Unless we also log known_version somewhere? (I'm just wondering if we'll be able to discern all the cases using metrics -- but feel free to disregard my comment! 😄)

Reply:
committed <= synced <= target <= highest

and then in the code as it is today

committed <= synced < target <= highest

because of the way the counters are hit I don't think nuance would be worth it whereas folks would always expect target to be such that synced < target <= highest otherwise what are you going after?

Reply:
Okay, this makes sense to me. I wasn't looking for anything specific -- just wanted to call out that if there is nuance in the metrics around edge cases, we should pay attention to them. But it sounds like there isn't anything interesting 😄